lib/logstash/util/compress.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/environment"
+
+module LogStash::Util
+  module Zip
+
+    extend self
+
+    def extract(source, target)
+      require "zip"
+      require "fileutils"
```

Comment:
I did the lazy require as we do it in so many places to load libraries, I tend to put everything on top. Can you educate me when is ok to be on top or when to do lazy require? thanks a lot.